src/difflib.h

```diff
@@ -91,30 +86,48 @@ template <class T> class has_bracket_operator {
   static bool const value = (sizeof(matcher<T>(nullptr)) == sizeof(has_op));
 };
 
-template <class T = std::string> class SequenceMatcher {
+/*
+ * taken from https://stackoverflow.com/a/17251989/11335032
+ */
+template <typename T, typename U> bool CanTypeFitValue(const U value)
+{
+    const intmax_t botT = intmax_t(std::numeric_limits<T>::min());
+    const intmax_t botU = intmax_t(std::numeric_limits<U>::min());
+    const uintmax_t topT = uintmax_t(std::numeric_limits<T>::max());
+    const uintmax_t topU = uintmax_t(std::numeric_limits<U>::max());
+    return !((botT > botU && value < static_cast<U>(botT)) ||
+             (topT < topU && value > static_cast<U>(topT)));
+}
+
+template <class T = std::string, class U = T> class SequenceMatcher {
   static_assert(is_standard_iterable<T>::value, "The matched objects must be iterable.");
   static_assert(is_hashable_sequence<T>::value, "The matched sequences must be of hashable elements.");
   static_assert(has_bracket_operator<T>::value, "The matched sequences must implement operator[].");
+  static_assert(is_standard_iterable<U>::value, "The matched objects must be iterable.");
+  static_assert(is_hashable_sequence<U>::value, "The matched sequences must be of hashable elements.");
+  static_assert(has_bracket_operator<U>::value, "The matched sequences must implement operator[].");
 
  public:
-  using value_type = T;
-  using hashable_type = typename T::value_type;
+  //using value_type = T;
```

Comment:
I am not really sure what would be a reasonable value_type when T and U are different types